skcosmo/preprocessing/FlexibleScaler.py

```diff
@@ -0,0 +1,242 @@
+
+            if self.column_wise:
+                if np.any(var == 0):
+                    raise ValueError("Cannot normalize a feature with zero variance")
+                self.scale_ = np.sqrt(var)
+            else:
+                var_sum = var.sum()
+                if var_sum == 0:
+                    raise ValueError("Cannot normalize a matrix with zero variance")
+                self.scale_ = np.sqrt(var_sum)
+
+        return self
+
+    def transform(self, X, y=None):
+        """Normalize a vector based on previously computed mean and scaling.
+
+        :param X: Matrix
+        :type X: ndarray
+        :param y: ignored
+        :return: transformed matrix X
+        """
+
+        if self.n_samples_seen_ == 0:
+            raise sk.exceptions.NotFittedError(
+                "This "
+                + type(self).__name__
+                + " instance is not fitted yet. Call 'fit' with appropriate arguments before using this estimator."
+            )
+        if (self.n_samples_seen_ != X.shape[0]) or (self.n_features_ != X.shape[1]):
+            raise ValueError("X shape does not match training shape")
+        return (X - self.mean_) / self.scale_
+
+    def fit_transform(self, X, y=None, **fit_params):
+        r"""Fit to data, then transform it.
+
+        :param X: Matrix
+        :type X: ndarray
+        :param y: ignored
+        :param \**fit_params: necessary for compatibility with the functions of the TransformerMixin class
+        :return: itself
+        """
+        self.fit(X, y)
+        return self.transform(X, y)
+
+    def inverse_transform(self, X_tr):
+        """Scale back the data to the original representation
+
+        :param X_tr: Matrix
+        :type X_tr: ndarray
+        :return: original matrix X
+        """
+        if self.n_samples_seen_ == 0:
+            raise sk.exceptions.NotFittedError(
+                "This "
+                + type(self).__name__
+                + " instance is not fitted yet. Call 'fit' with appropriate arguments before using this estimator."
+            )
+        if (self.n_samples_seen_ != X_tr.shape[0]) or (
+            self.n_features_ != X_tr.shape[1]
+        ):
+            raise ValueError("X shape does not match training shape")
+        return X_tr * self.scale_ + self.mean_
+
+
+class KernelFlexibleCenterer(TransformerMixin, BaseEstimator):
+    """Kernel centering method, similar to KernelCenterer,
+    but with additional parameters, relative to which centering
+    is carried out:
+
+    :param ref_cmean: an array with means for each column.
+    :type ref_cmean: array of shape (insert shape here)
+
+    :param ref_rmean: an array with means for each row.
+    :type ref_rmean: array of shape (insert shape here)
+
+    :param ref_mean: an average for the whole kernel matrix
+    :type ref_mean: array
+    """
+
+    def __init__(self, *, ref_cmean=None, ref_rmean=None, ref_mean=None):
```

Comment:
```suggestion
    def __init__(self):
```

Reply:
We need to declare it. Otherwise, a convenient check that we have not yet made a fit does not work.